### ISSUE TITLE
Framebuffers on iOS stopped working (version 1.9.x)

### ISSUE THREAD

**Issue by orangepascal** [libgdx/libgdx#3864]
#### Issue details
Rendering on iOS (any version) worked in 1.7.x, now upgraded to 1.9.x and it fails (haven't tried 1.8.x)
Issue is related to framebuffers, without using an FBO it renders normally.

#### Reproduction steps/code
Small example code: https://bitbucket.org/orangepixel/tiny-puncher

with Libgdx 1.7 it runs normal on iOS and shows everything, if you upgrade it to 1.9 it will run normal but the screen stays black.  (removing the rendering to a framebuffer from com.orangepixel.utils.arcadecanvas render()  makes it work again)

problem seems to be in framebuffers, possible the cause of the problem:
https://github.com/libgdx/libgdx/commit/ef0c210e9a26892e5b55baf2abd23bd60d13f448

#### Version of LibGDX and/or relevant dependencies
1.7.x works,  1.9.x doesn't

#### Please select the affected platforms
- [ ] Android
- [X] iOS
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [ ] MacOS

**Comment by davebaol** [comment]
> Small example code: https://bitbucket.org/orangepixel/tiny-puncher

Well, not quite the small example everyone is thinking of :smile_cat:

**Comment by xoppa** [comment]
Yeah, can you be a bit more specific @orangepascal, perhaps try the [FrameBufferTest](https://github.com/libgdx/libgdx/blob/master/tests/gdx-tests/src/com/badlogic/gdx/tests/FrameBufferTest.java) or any of the other tests that use fbo. Framebuffer worked on iOS in my tests after that commit, so I doubt that Framebuffer as whole is not working on iOS because of that. Can't test at the moment though.

**Comment by orangepascal** [comment]
ok ok the basic version  :)
https://bitbucket.org/snippets/orangepascal/dKnGp

my ipad is running ios 8 and this won't render on it with 1.9.x, but does with 1.7.  
Disable the use of the framebuffer, and it does render

**Comment by xoppa** [comment]
Is there a particular reason you are creating the fbo in the resize method and using a TextureRegion instead of the texture directly? See #3850

**Comment by orangepascal** [comment]
normally I have a framebuffer in "low-res" pixelart, based on the size of the window for correct aspect ratio's.. so I create it in resize to make sure it scales correctly on all systems.  Knew about that bug, fixed that, like I mentioned the game does run, just doesn't render.

also normally I use some lights or filters on top  rendering to the main framebuffer before I then scale that framebuffer to the full screen.. 

basically everything I did in the project I posted before, but you wanted the bare bone problem case :) 

which is what I posted now :)

**Comment by xoppa** [comment]
So not using the TextureRegion and resize method also shows the problem? As said, I can't test at the moment, so just trying to help by looking at the code.

**Comment by orangepascal** [comment]
yup, updated the snippet to just the framebuffer usage:
https://bitbucket.org/snippets/orangepascal/dKnGp

the problem really is in framebuffer :)

**Comment by orangepascal** [comment]
oh, when not using the onResize it does work (even tho onresize gives the correct width+height on ios)

**Comment by badlogic** [comment]
Hm, might mean no OpenGL context is active when onResize is called.

**Comment by xoppa** [comment]
Yeah the `resize` issue needs to be fixed, it causes quite some problems (e.g. `viewport.update` won't work when the viewport isn't created yet). Unfortunately I can't verify the fix referenced in #3850, perhaps @badlogic or someone else can have a look at it.

**Comment by jodik** [comment]
I have exactly the same issue (just on iOS). FrameBuffer works in 1.7.0 and in 1.9.2 nothing is rendered once I call FrameBuffer.begin(). Is it fixed with a7e7f5c?

**Comment by orangepascal** [comment]
the problem is indeed still there.
onResize is correctly called after the onCreate, but the problem (with Framebuffer) is still there. Nothing being rendered

**Comment by Tom-Ski** [comment]
Can supply a simple ApplicationAdapter that reproduces your problem so we can test with a minimal setup

----- Reply message -----

**Comment by orangepascal** [comment]
I wrote a little bare-bones test-case, but it works in that one.
tried again in my own code and it doesn't there.

Moved the creation of my framebuffers outside the resize() call, and then it works.

Guessing my own code has a longer init (loading graphics,sounds,etc) causing the difference between that and the bare-bones test-case? possibly

still not a perfect solution, cause you normally want to recreate framebuffers after a resize().  
For now I just dispose them in the resize, and they will be recreated on the next render call.. messy, but I'll take it for now.

so something goes wrong between framebuffers and ios' way of handling the resize (which does trigger 4 times during startup/application create )

**Comment by Tom-Ski** [comment]
Do you only create your FBOs in resize, or do you also create in `create`?  Is this still happening with TinyPuncher? If so I can take a look tonight.

**Comment by orangepascal** [comment]
I tried creating both in create+resize and just in resize which failed.. creating them in the resize basically just fails.

just tested TinyPuncher version, and that does work!   I did notice one difference between the cases which I can't find anything that causes it.  The working cases startup with this console output:

IOSApplication: iOS version: 8.1.1
IOSApplication: Running in 32-bit mode
IOSApplication: Unscaled View: LandscapeRight 1024x768
IOSApplication: View: LandscapeRight 1024x768
IOSGraphics: 1024.0x768.0, 1.0
IOSGraphics: Display: ppi=132, density=0.825
IOSApplication: created
IOSApplication: resumed		


and the code going wrong does this:

IOSApplication: iOS version: 8.1.1
IOSApplication: Running in 32-bit mode
IOSApplication: Status bar is not visible
IOSApplication: Total computed bounds are w=1024.0 h=768.0
IOSGraphics: Display: ppi=132, density=0.825
IOSApplication: created
IOSApplication: Status bar is not visible
IOSApplication: Total computed bounds are w=1024.0 h=768.0
IOSApplication: Status bar is not visible
IOSApplication: Total computed bounds are w=1024.0 h=768.0


I think it's roboVM output, but both versions use libgdx 1.9.2 and robovm 1.9.0.. so I'm not really sure where the difference comes from and if this might be the cause of it?

**Comment by arkiruthis** [comment]
I've run into exactly this same issue (Gdx 1.9.2, RoboVM 1.9.0).

In create() I glClear() a framebuffer to red. On every platform (apart from iOS) I get a red texture back. On iOS I just get black. Doesn't seem to be just a resize() issue here, I get a black texture from the framebuffer regardless of where I try to create it on iOS.

I notice the ticket has been marked as "Closed" but there didn't appear to be a solution?

**Comment by Tom-Ski** [comment]
The original issue originated as resize was being called in the wrong place.  It was closed as that issue should have been resolved.  There hasn't been an self contained example that has shown this to still be a problem, if you can provide one we can work from there.

**Comment by arkiruthis** [comment]
Thanks Tom-Ski, if I'm able to replicate it in a fresh minimal project I'll create a new issue.

**Comment by Tom-Ski** [comment]
Looks like they are running different versions, that first console output startup is pre 1.9.2

**Comment by orangepascal** [comment]
ah gotcha.. seems the test cases were running on 1.7.1 indeed.  Updating them to 1.9.2 gives us the problem.

So here's a short example of the problem:
https://bitbucket.org/orangepascal/libgdxframebuffertest

**Comment by Tom-Ski** [comment]
Your example works OK for me. Running on iOS 9.3 and latest source of LibGDX.  This could be an iOS version issue.  Could you try this ApplicationAdapter https://gist.github.com/Tom-Ski/33d7f8819aca782ad55626bad7e1ef45

**Comment by orangepascal** [comment]
yeah that works, it's basically the work-around I use in my game now.
I'm testing on ios 8.1.1
